velox/functions/prestosql/aggregates/ChecksumAggregate.cpp

```diff
@@ -0,0 +1,228 @@
+      folly::Range<const vector_size_t*> indices) override {
+    for (auto i : indices) {
+      *value<int64_t>(groups[i]) = 0;
+    }
+  }
+
+  void extractValues(char** groups, int32_t numGroups, VectorPtr* result)
+      override {
+    auto* vector = (*result)->as<FlatVector<StringView>>();
+    VELOX_CHECK(vector);
+    vector->resize(numGroups);
+
+    auto* rawValues = vector->mutableRawValues();
+    for (vector_size_t i = 0; i < numGroups; ++i) {
+      auto group = groups[i];
+      clearNull(group);
+      rawValues[i] = StringView(value<char>(group), sizeof(int64_t));
+    }
+  }
+
+  void extractAccumulators(char** groups, int32_t numGroups, VectorPtr* result)
+      override {
+    auto vector = (*result)->as<FlatVector<int64_t>>();
+    VELOX_CHECK(
+        vector,
+        "Unexpected type of the result vector: {}",
+        (*result)->type()->toString());
+    vector->resize(numGroups);
+
+    for (int32_t i = 0; i < numGroups; ++i) {
+      auto group = groups[i];
+      if (isNull(group)) {
+        vector->setNull(i, true);
+      } else {
+        vector->set(i, *value<int64_t>(group));
+      }
+    }
+  }
+
+  void addRawInput(
+      char** groups,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool /**/) override {
+    const auto& arg = args[0];
+    auto hasher = getPrestoHasher(arg->type());
+    auto hashes = getHashBuffer(rows.end(), arg->pool());
+    hasher->hash(arg, rows, hashes);
+    auto rawHashes = hashes->as<int64_t>();
+
+    rows.applyToSelected([&](vector_size_t row) {
+      if (arg->isNullAt(row)) {
+        computeHashForNull(groups[row]);
+      } else {
+        computeHash(groups[row], rawHashes[row]);
+      }
+    });
+  }
+
+  void addIntermediateResults(
+      char** groups,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool /**/) override {
+    auto vector = args[0]->asUnchecked<FlatVector<int64_t>>();
+    auto rawValues = vector->rawValues();
+
+    rows.applyToSelected(
+        [&](vector_size_t i) { *value<int64_t>(groups[i]) += rawValues[i]; });
+  }
+
+  void addSingleGroupRawInput(
+      char* group,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool /**/) override {
+    const auto& arg = args[0];
+    auto hasher = getPrestoHasher(arg->type());
+    auto hashes = getHashBuffer(rows.end(), arg->pool());
+    hasher->hash(arg, rows, hashes);
+    auto rawHashes = hashes->as<int64_t>();
+
+    rows.applyToSelected([&](vector_size_t row) {
+      if (arg->isNullAt(row)) {
+        computeHashForNull(group);
+      } else {
+        computeHash(group, rawHashes[row]);
+      }
+    });
+  }
+
+  void addSingleGroupIntermediateResults(
+      char* group,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool /**/) override {
+    clearNull(group);
+
+    auto vector = args[0]->asUnchecked<FlatVector<int64_t>>();
```

Comment:
consistency: other places use temp variable for args[0] and check that vector is not null